### PR TITLE
Extract $releaseDatetime in order to share a common release datetime between parallel deployments.

### PR DESCRIPTION
**Pull request by tomzx**

### CODE REVIEW

File: recipe/common.php [hunk 1]
```diff
@@ -71,16 +71,10 @@
...
-
+$release = date('YmdHis');
+task('deploy:release', function () use ($release) {
     $releasePath = "{deploy_path}/releases/$release";
 
-    $i = 0;
-    while (is_dir(env()->parse($releasePath)) && $i < 42) {
-        $releasePath .= '.' . ++$i;
-    }
-
```
Comment (elfet) [inline recipe/common.php]: Why you delete this?

**Comment by elfet** [comment]
Yes, this is good feature. But it will not working in parallel mode. I will implement distributed Deployer parameters, so it will be possible to implement correct solution for this in parallel deployment mode.

File: recipe/common.php (hunk 1 above)
Comment (tomzx) [inline recipe/common.php]: Is it really needed? And why stop at 42?

Comment (elfet) [inline recipe/common.php]: Yes, it's needed in tests. Tree time runs `deploy:release` and check all was correct. 
[42](https://simple.wikipedia.org/wiki/42_(answer))

**Comment by elfet** [comment]
Build fails.

**Comment by tomzx** [comment]
@elfet With the latest change, I've created an global variable called `release_identifier`, which is more generic than before, and allows users to redefine how they want their release identifier to work so they could use the old numbering system if they prefer it).

**Comment by KingCrunch** [comment]
Or the VCSs revision/hash. Just saying :smile:

**Comment by tomzx** [comment]
If I'm not mistaken, the way Capistrano does it is to append the release at the end of a file containing the list of releases. Whatever the identifier is, it'll be possible to figure out which release came before/after which one based on the ordering in the file. It shouldn't be too hard to implement this feature if enough people want it.

**Comment by KingCrunch** [comment]
Just tried it myself: Capistrano3 keeps a log of revision it deployed, but that seems to be just a log.

```bash
$ cat revisions.log 
Branch master (at bdd236b) deployed as release 20150318214035 by sebastian
```
It still creates the actual deployments into folders based on the timestamp and delete them in order. However, that doesn't me `deployer` can't do it differently :smile:

**Comment by elfet** [comment]
Will be implemented in Deployer v4.

**Pull request closed without merging**